com.unity.sg2/Editor/GraphUI/DataModel/ShaderGraphModel.cs

```diff
@@ -31,93 +32,40 @@ class MainPreviewData
         [NonSerialized]
         public float scale = 1f;
 
-        public void Initialize()
+        public Mesh mesh
         {
-            if (serializedMesh.IsNotInitialized)
-            {
-                // Initialize the sphere mesh as the default
-                Mesh sphereMesh = Resources.GetBuiltinResource(typeof(Mesh), $"Sphere.fbx") as Mesh;
-                serializedMesh.mesh = sphereMesh;
-            }
+            get => serializedMesh.mesh;
+            set => serializedMesh.mesh = value;
         }
     }
 
     public class ShaderGraphModel : GraphModel
     {
-        public GraphHandler GraphHandler => ShaderGraphAssetModel.GraphHandler;
-
-        public ShaderGraphAssetModel ShaderGraphAssetModel => Asset as ShaderGraphAssetModel;
-
-        public Registry RegistryInstance => ((ShaderGraphStencil)Stencil).GetRegistry();
-
-        #region MainPreviewData
-        MainPreviewData m_MainPreviewData = new ();
-
-        internal MainPreviewData mainPreviewData => m_MainPreviewData;
-
-        public void SetPreviewMesh(Mesh newPreviewMesh)
+        [SerializeField]
+        private SerializableGraphHandler graphHandlerBox = new();
+        [SerializeField]
+        private SerializableTargetSettings targetSettingsBox = new();
+        [SerializeField]
+        private MainPreviewData mainPreviewData = new();
+        [SerializeField]
+        private bool isSubGraph = false;
+
+        internal GraphHandler GraphHandler => graphHandlerBox.Graph;
+        internal Registry RegistryInstance => graphHandlerBox.Graph.registry;
+        internal List<JsonData<Target>> Targets => targetSettingsBox.Targets;
+        internal MainPreviewData MainPreviewData => mainPreviewData;
+        internal bool IsSubGraph => CanBeSubgraph();
+
+        public void Init(GraphHandler graph, bool isSubGraph)
```

Comment:
Init has been refactored to only get called when the graph is first initialized. It's serialization state should be valid from that point on that it no longer needs any special handling or management. There is a TODO here to provide an initial `TargetSettingsObject`, or the equivalent as well-- as we'll have various `Create` functions that initialize a graph based on any number of things.